src/PrestaShopBundle/Form/Admin/ShopParameters/General/MaintenanceFormDataProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Form\Admin\ShopParameters\General;
+
+use PrestaShop\PrestaShop\Adapter\Shop\MaintenanceConfiguration;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+/**
+ * This class is responsible of managing the data manipulated using forms
+ * in "Configure > Shop Parameters > General > Maintenance" page.
+ */
+class MaintenanceFormDataProvider
```

Comment:
I see you have merged the PR which introduces `FormDataProviderInterface`, rebase your pr and use it, and you can use `{@inheritdoc}` annotations too :+1:

Reply:
Also, form data providers are usually declared as `final`: if someone want to change the behavior, he'll need to implement the interface instead of re-use the classes we will unit test (in the end)